src/index/CompressedRelation.cpp

```diff
@@ -729,20 +732,19 @@ std::pair<size_t, size_t> CompressedRelationReader::getResultSizeImpl(
 
 // ____________________________________________________________________________
 std::pair<size_t, size_t> CompressedRelationReader::getSizeEstimateForScan(
-    const ScanSpecification& scanSpec,
-    const vector<CompressedBlockMetadata>& blocks,
+    const ScanSpecification& scanSpec, BlockMetadataRanges blockRanges,
     const LocatedTriplesPerBlock& locatedTriplesPerBlock) const {
-  return getResultSizeImpl<false>(scanSpec, blocks, locatedTriplesPerBlock);
+  return getResultSizeImpl<false>(scanSpec, blockRanges,
+                                  locatedTriplesPerBlock);
```

Comment:
When you implement my changes from above, then the `joinedViews` (whatever their type is), 
can again be called `blocks`, to minimize the diff here.